cpu/atmega328p/Kconfig

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2020 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+config CPU_SERIES_ATMEGA328
+    bool
+    select CPU_FAMILY_ATMEGA
+    select HAS_ATMEGA_PCINT1
+    select HAS_ATMEGA_PCINT2
+
+## CPU Models
+config CPU_MODEL_ATMEGA328P
+    bool
+    select CPU_SERIES_ATMEGA328
+    select HAS_CPU_ATMEGA328P
+
+## Definition of specific features
+config HAS_CPU_ATMEGA328P
+    bool
+    help
+        Indicates that a 'atmega328p' cpu is being used.
+
+## Common CPU symbols
+config CPU_SERIES
+    default "atmega328" if CPU_SERIES_ATMEGA328
+
+config CPU_MODEL
```

Comment:
Is it actually needed to have `CPU_MODEL` and `CPU_MODEL_ATMEGA328P`, could you have one and assign it in the `BOARD` kconfig file?

```
config CPU_MODEL
    default "atmega328p"
```

Reply:
Is it mainly to make `if` satements easier?

Reply:
Generally it's easier to check for boolean symbols in expressions

Reply:
But in the end it seems to lead to duplicated symbols for everything right? Is this a common practice in kconfig? If it is I have nothing against it, but its something I would like documented, since this does not seems evident to me.

Reply:
Do you mean declaring the multiple defaults?

Reply:
Keep in mind that the declarations of symbols in `cpu/Kconfig` are main ones which assign the help string and the type. All the others are just extending the definitions with conditional default values, which is a common practice AFAIK

Reply:
I mean that for a lot of the variables you end up with `CPU_SERIES` that has a default value `foo` and then also `CPU_SERIES_FOO`. But maybe this is very evident when you use kconfig a lot? But if this was my first time doing a board `kconfig` file I would have just done:

```
config CPU_MODEL
    default "atmega328p"
```

Reply:
It's somehow similar to what we have now in `riotbuild.h`:
```
#define BOARD_NATIVE "native"
#define RIOT_BOARD BOARD_NATIVE
```
I see that the boolean symbols are good for use in expressions and then the string symbols could be use in the application or build system.

Reply:
I understand, but I think this should somehow be a convention then. Current documentation is very configuration based, and It would be better to extend it towards how to strucutre `FEATURES` declarations. Does that make sense?

Reply:
I will document this.

Reply:
Will be addressed in follow up